### PR TITLE
Created "/v2/onset" endpoint

### PR DESCRIPTION
**Pull request by lmeinen**
In order to upload both to the red (SwissCovid) and the purple (NotifyMe) backend, a second token needs to be returned for a single authorization code. To this end, the "/v2/onset" (cf. "v1/onset") endpoint was created.

This involved:
- Creating a second Controller class `AuthCodeVerificationControllerV2`
- Modifying the `AuthCodeVerificationService` to accept a boolean parameter, indicating whether a second token should be returned.
- Adding a `TokenType` enum with `audience` and `scope` parameters, which is used in the `CustomTokenProvider`'s `createToken` method to set `aud` and `scope` accordingly.
- Creating the `AuthorizationCodeVerifyResponseDtoWrapper` class to wrap the two returned access tokens.

Note that the dp3t backend currently doesn't check the `aud` claim of the passed JWT token. To avoid the possibility being reused for the red backend, the `scope` claimn of the NotifyMe access token needs to be different.

### CODE REVIEW

File: src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  public AuthorizationCodeVerifyResponseDtoWrapper(AuthorizationCodeVerifyResponseDto swissCovidAccessToken, AuthorizationCodeVerifyResponseDto notifyMeAccessToken) {
+    this.swissCovidAccessToken = swissCovidAccessToken;
+    this.notifyMeAccessToken = notifyMeAccessToken;
+  }
+
+  public AuthorizationCodeVerifyResponseDtoWrapper() {}
+
+
+  public AuthorizationCodeVerifyResponseDto getSwissCovidAccessToken() {
```
Comment (UBaggeler) [inline src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java]: Both tokens are used for SwissCovid. I would rename this to `getDP3TAccessToken`.

File: src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    return swissCovidAccessToken;
+  }
+
+  public void setSwissCovidAccessToken(AuthorizationCodeVerifyResponseDto swissCovidAccessToken) {
+    this.swissCovidAccessToken = swissCovidAccessToken;
+  }
+
+  public AuthorizationCodeVerifyResponseDto getNotifyMeAccessToken() {
```
Comment (UBaggeler) [inline src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java]: I'd suggest to rename this to  `getCheckInUserUploadAccessToken`.

File: src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/TokenType.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package ch.admin.bag.covidcode.authcodegeneration.api;
+
+public enum TokenType {
+    SWISSCOVID_TOKEN("swissCovid", "exposed"), NOTIFYME_TOKEN("notifyMe", "userupload");
```
Comment (UBaggeler) [inline src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/TokenType.java]: How about `DP3T_TOKEN("dp3t", "exposed"), CHECKIN_USERUPLOAD_TOKEN("checkin", "userupload");`?

**Comment by UBaggeler** [comment]
I did a first pass. Apart from some naming changes (trying to use "checkin" and "dp3t" to avoid confusion) I think this is the way to go.

File: src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/TokenType.java (hunk 3 above)
Comment (martinalig) [reply src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/TokenType.java]: done

File: src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java (hunk 2 above)
Comment (martinalig) [reply src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java]: done

File: src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java (hunk 1 above)
Comment (martinalig) [reply src/main/java/ch/admin/bag/covidcode/authcodegeneration/api/AuthorizationCodeVerifyResponseDtoWrapper.java]: done

**Review by UBaggeler — APPROVED** [review]

**Pull request merged**